models/interventions.py

```diff
@@ -253,6 +253,58 @@ def __str__(self):
         return f"BoundlessRotatedSpaceIntervention(embed_dim={self.embed_dim})"
 
 
+class SigmoidMaskRotatedSpaceIntervention(TrainableIntervention):
+
+    """Intervention in the rotated space with boundary mask."""
+    def __init__(self, embed_dim, **kwargs):
+        super().__init__(**kwargs)
+        rotate_layer = RotateLayer(embed_dim)
+        self.rotate_layer = torch.nn.utils.parametrizations.orthogonal(
+            rotate_layer)
+        self.subspace_partition = kwargs["subspace_partition"] \
+            if "subspace_partition" in kwargs else None
+        # TODO: in case there are subspace partitions, we
+        #       need to initialize followings differently.
+        self.masks = torch.nn.Parameter(
```

Comment:
it maybe better to initialize this such that all bits in the mask are turned on initially. the idea is that we want to push down the number of bits being turned on through an external loss such as `mask.sum()`.